flake.nix

```
@@ -0,0 +1,57 @@
+{
+  description = "cdk-validium-node, with go development environment";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+    flake-compat.url = "github:edolstra/flake-compat";
+    flake-compat.flake = false;
+    pre-commit-hooks.url = "github:cachix/pre-commit-hooks.nix";
```

Should we use input follows?
```suggestion
    pre-commit-hooks = { 
      url = "github:cachix/pre-commit-hooks.nix";
      inputs = {
        nixpkgs.follows = "nixpkgs";
        flake-compat.follows = "flake-compat";
      };
    };
```

---

sure, we can use consistent version of `nixpkgs` and `flake-compat`.